docs/rfc/0002-identity_proofs.md

```diff
@@ -0,0 +1,212 @@
+
+To lift the requirement for physical authenticity checks, but still increase
+confidence of a given public key being associated with a particular person,
+[keybase] have popularised a scheme dubbed "Social Proofs": a statement claiming
+ownership of a particular account on a social media site is written to the
+keybase "sigchain" (which has properties similar to a `radicle-link` identity).
+This claim (implying also the history of the sigchain) is signed using a key
+currently valid according to the chain, and the signature (along with the public
+key) is stored at the social media site. To verify the claim, the signature is
+retrieved from the social media site in such a way that it could _plausibly_
+only be created by the owner of the claimed account. If both the sigchain
+integrity and the claim signature can be verified, the association is proven.
+
+This mechanism can be considered a practical application of the [Turing
+test][tt]: even though it can not be proven beyond doubt that the account is
+indeed associated with a real person, the evidence of others accepting it as
+such, as well as conversational behaviour, can increase the confidence in the
+authenticity of the online persona. Because key and account ownership at a given
+point in time can be cryptographically verified, this confidence can be extended
+to the proving side.
+
+We conclude that this mechanism would be a good fit for `radicle-link` (due to
+the similarities), and a desirable feature of applications built on top of it.
+
+The following sections describe how such claims shall be stored in the identity
+payload of a `radicle-link` identity, how to obtain a publishable proof, and how
+to verify it.
+
+## Claims
+
+Claims can only be made by identities of kind `Person`, and claim a single
+external account identifier. They are introduced by defining a new payload type,
+identified by the URL:
+
+    https://radicle.xyz/link/claim/v1
+
+The shape of the JSON structure is:
+
+```json
+{
+    "SERVICE": {
+        "account": "STRING"
+        "expiration": {
+            "created": INTEGER,
+            "expires": INTEGER
+        },
+        "proof": "URL"
+    }
+}
+```
+
+Where the fields denote:
+
+* `SERVICE`
+
+  A conventional identifier of the external service, e.g. "github", "twitter",
+  "radicle-ethereum"
+
+* `account`
+
+  The unique account identifier within the service, using the service-specific
+  canonical string representation e.g. "kim", "0x32be343b94f860124dc4fee278fdcbd38c102d88".
+
+* `expiration` (optional)
+    * `created`
+
+      Creation timestamp of the claim, in seconds since 1970-01-01T00:00:00Z.
+
+    * `expires`
+
+      Seconds relative to `created`, after which the claim should no longer be
+      considered.
+
+* `proof` (optional)
+
+  A URL to assist verification tooling in retrieving the proof from the external
+  system. This is mainly a convenience, and obviously requires creation of a new
+  revision after the fact.
```

Comment:
Why after the fact?

Reply:
This almost seems like a non-feature, since it requires a new identity revision, I'm not sure if clients will deem it worthwhile to support :shrug:

Reply:
Since there’s no central database, this would mean that you can only navigate from the external system to link, but not the other way round — which would be a bit lame I think.

Perhaps the proof signature could be over the previous revision + a random nonce 🤔 If space is constrained, maybe the preimage could be omitted entirely (except the root hash)?

Reply:
Yeah, true.

Signing the previous revision + account + nonce is not a bad idea :thinking: but seems like it may complicate implementations slightly due to the verification requiring two revisions as input instead of one?

I'm wondering in what scenarios the proof URL will be used? Because if there are legit use-cases for it, then it probably shouldn't be optional, and we should go with your idea.

So the typical use-case is that of sybil.org for example, where you can verify the statement: "I am @cloudhead and address 0xabcdef123 belongs to me". They must store the tweet id somewhere, since that's how they verify the signature. So yeah it seems kind of important :thinking:

Reply:
> the verification requiring two revisions as input instead of one?

Myeah, so:

## With Nonce

1. Post signature over `R || nonce`, store `URL`
2. Create `R+1` containing `nonce`, `SERVICE: {..., proof: Some(URL) }`
3. Sign `R+1` using `delegations` to "finalise" it

### Verify

1. Check signature, abort if invalid
2. Find `R+1`, check that
    1. is valid
    2. `R` is parent
    3. `R` has key in delegations
    4. `R+1` has claim
    5. `R+1` claim is not expired
3. Check `HEAD` still has the claim

vs.

## Without Nonce

1. Create `R` containing `SERVICE: { ..., proof: None }`
2. Post signature over `R`, store `URL`
3. Create `R+1` with `SERVICE: { ..., proof: Some(URL) }`
4. Sign `R` **and** `R+1` using `delegations` to finalise

### Verify

1. Check signature, abort if invalid
2. Find `R`, check that
    1. is valid
    2. has key in delegations
    3. has claim
    4. claim is not expired
3. Check `HEAD` still has the claim
  
So the issue with the nonce is that you need to publish the nonce, _or_ that you can only verify the signature if you know `R+1`'s _payload_. The issue without the nonce is that you need to sign two successive revisions for finalisation.

> I'm wondering in what scenarios the proof URL will be used?

It's when you start from `link` -- you'll want an O(1) method of getting at the proof payload.

> then it probably shouldn't be optional

Without a nonce, it's a chicken-and-egg problem :)


> sybil.org for example

sybil.org is simpler, because:

1. there is only **one** key (the address)
2. they basically store _all_ claims in a central database ([verified.json](https://github.com/Uniswap/sybil-list/blob/master/verified.json))
3. there is nothing to verify about the database itself (because 1.) 

I mean, the proposed proof URL doesn't technically need to be signed, either. But if it's not in the identity document, then where else could we put it? Annotate `R` with a [git-notes](https://git-scm.com/docs/git-notes) note?